src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java

```diff
@@ -92,14 +97,42 @@ public HttpCassandraManagementProxy(MetricRegistry metricRegistry,
                                       String rootPath,
                                       InetSocketAddress endpoint,
                                       ScheduledExecutorService executor,
-                                      DefaultApi apiClient
+                                      DefaultApi apiClient,
+                                      int metricsPort,
+                                      Node node
   ) {
     this.host = endpoint.getHostString();
     this.metricRegistry = metricRegistry;
     this.rootPath = rootPath;
     this.endpoint = endpoint;
     this.apiClient = apiClient;
+    this.metricsPort = metricsPort;
     this.statusTracker = executor;
+    this.node = node;
+    this.metricsProxy = HttpMetricsProxy.create(this, node);
```

Comment:
Suggestion: OK, this feels a bit weird. Our MetricsProxy takes a `HttpCassandraManagementProxy` as a argument to the constructor, but `HttpCassandraManagementProxy` also has an instance of the `MetricsProxy`, it feels a bit tangled. Is it possible to get some clearer separation between these classes?

Possibly this is an item for later refactoring, but I had to ask...

Reply:
It actually looks like the proxy is called only for this line of code:

```
String url = String.format("http://%s:%s/metrics", proxy.getHost(), proxy.getMetricsPort());
```

It seems to me like maybe the HttpCassandraManagementProxy shouldn't know about the metrics port at all. And the MetricsProxy should maybe take the host as an argument to it's constructor instead of taking the whole ICassandraManagementProxy.

I haven't looked at what this would imply for the JMX side of things, but think we should try to do something a bit cleaner here.

Reply:
Alex has said he's creating a refactoring ticket.